src/hooks/useHtmlPaste/index.ts

```diff
@@ -0,0 +1,129 @@
+import {useNavigation} from '@react-navigation/native';
+import ExpensiMark from 'expensify-common/lib/ExpensiMark';
+import {useCallback, useEffect} from 'react';
+import type UseHtmlPaste from './types';
+
+const useHtmlPaste: UseHtmlPaste = (textInputRef, checkComposerVisibility, isUnmountedOnBlur = true) => {
+    const navigation = useNavigation();
+
+    /**
+     * Set pasted text to clipboard
+     * @param {String} text
+     */
+    const paste = useCallback((text: string) => {
+        try {
+            document.execCommand('insertText', false, text);
+            // Pointer will go out of sight when a large paragraph is pasted on the web. Refocusing the input keeps the cursor in view.
+            textInputRef.current?.blur();
+            textInputRef.current?.focus();
+            // eslint-disable-next-line no-empty
+        } catch (e) {}
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    /**
+     * Manually place the pasted HTML into Composer
+     *
+     * @param {String} html - pasted HTML
+     */
+    const handlePastedHTML = useCallback(
+        (html: string) => {
+            const parser = new ExpensiMark();
+            paste(parser.htmlToMarkdown(html));
+        },
+        [paste],
+    );
+
+    /**
+     * Paste the plaintext content into Composer.
+     *
+     * @param {ClipboardEvent} event
+     */
+    const handlePastePlainText = useCallback(
+        (event: ClipboardEvent) => {
+            const plainText = event.clipboardData?.getData('text/plain');
+            if (plainText) {
+                paste(plainText);
+            }
+        },
+        [paste],
+    );
+
+    const handlePaste = useCallback(
+        (event: ClipboardEvent) => {
+            if (!textInputRef.current) {
+                return;
+            }
+            const isVisible = typeof checkComposerVisibility === 'function' && checkComposerVisibility();
```

Comment:
Maybe default to a function that returns `true` here and we can just call the function even when not passed.